### PR TITLE
Proxy providers system

### PR DESCRIPTION
**Pull request by Petrprogs**
Added proxy providers system. Some small enhancement/fixes

### CODE REVIEW

**Comment by Petrprogs** [comment]
@0n1cOn3  @walkero-gr What do you think of it?

File: main.py [hunk 1]
```diff
@@ -6,80 +6,138 @@
...
+            done > 3
+            and (downloaded_bytes // (time.perf_counter() - start_time) / 100000) < 1.0
+        ):
             print("\nProxy is too slow, skipping...")
-            return float('inf'), downloaded_bytes
-        sys.stdout.write("\r[%s%s] %s Mbps" % ('=' * done, ' ' * (30-done), downloaded_bytes//(time.perf_counter() -
-        start_time) / 100000))
+            return float("inf"), downloaded_bytes
+        if done > 5:
+            break
```
Comment (walkero-gr) [inline main.py]: Lines 84 and 85 might need to be above line 78, because a value i.e. 6 will into the above block of code and never reach the break.

**Comment by walkero-gr** [comment]
Nice implementation. Looking forward to testing it. Great job.

File: main.py (hunk 1 above)
Comment (Petrprogs) [reply main.py]: Fixed, thank you :)

**Comment by Petrprogs** [comment]
> Nice implementation. Looking forward to testing it. Great job.

Thank you for you review!

**Comment by Petrprogs** [comment]
Let's merge it!

**Review by 0n1cOn3 — APPROVED** [review]
Looks pretty neat. 
Thanks for the request @Petrprogs .
If nothing stays in the way, merge it 😌

**Pull request merged**